packages/expo-print/ios/ExpoPrintWithPrinter.swift

```diff
@@ -137,11 +138,39 @@ public class ExpoPrintWithPrinter {
     return printInteractionController
   }
 
-  private func dataFromUri(uri: String) -> Data? {
-    do {
-      return try Data(contentsOf: URL(fileURLWithPath: uri))
-    } catch {
-      return nil
+  private func dataFromUri(uri: String, completion: @escaping (Data?) -> Void) {
+    if uri.hasPrefix("http") {
+      // Handle HTTP URL asynchronously
+      DispatchQueue.global().async {
+        if let url = URL(string: uri), let data = try? Data(contentsOf: url) {
+          DispatchQueue.main.async {
+            completion(data)
+          }
+        } else {
+          DispatchQueue.main.async {
+            completion(nil)
+          }
+        }
+      }
+    } else if uri.hasPrefix("data:") {
+      // Handle base64 encoded data URI
+      if let commaIndex = uri.firstIndex(of: ",") {
+        let base64String = String(uri[uri.index(after: commaIndex)...])
+        if let data = Data(base64Encoded: base64String) {
+          completion(data)
+        } else {
+          completion(nil)
+        }
+      }
+    } else {
+      // Handle local file path synchronously
+      let adjustedUri = uri.hasPrefix("file://") ? String(uri.dropFirst(7)) : uri
+
+      if let data = try? Data(contentsOf: URL(fileURLWithPath: adjustedUri)) {
+        completion(data)
+      } else {
+        completion(nil)
+      }
```

Comment:
URL can handle the data and "file://" prefixes so we can reduce the number of ifs. Also we should use `URLSession` for http requests. In the variable declarations at the top of the file we should add:
```
  let printURLSession = URLSession(configuration: .default)
  var dataTask: URLSessionDataTask?
```
Here is an improved version of the function:
```suggestion
   private func dataFromUri(uri: String, completion: @escaping (Data?) -> Void) {
    guard var url = URL(string: uri) else {
      completion(nil)
      return
    }
    
    // Assume that URLs without a scheme eq. /home/user/file.pdf will be local file urls
    if url.scheme == nil {
      url = URL(fileURLWithPath: uri)
    }
    
    // process http and https requests asynchronously
    if url.scheme == "http" || url.scheme == "https" {
      dataTask = printURLSession.dataTask(with: URLRequest(url: url)) { [weak self] data, response, error in
        defer {
          self?.dataTask = nil
        }
        
        if error == nil, let response = response as? HTTPURLResponse, response.statusCode == 200 {
          DispatchQueue.main.async {
            completion(data)
          }
          return
        }
        completion(nil)
      };
      dataTask?.resume()
    } else {
      let data = try? Data(contentsOf: url)
      completion(data)
    }
  }
```